spec/support/contexts/odk_submissions.rb

```diff
@@ -47,23 +47,23 @@ def build_odk_submission(form, data: {}, override_form_id: false, repeat: false,
               xml << "<grp#{item.id}><header/>"
               subitems.each do |subitem, _|
                 if instance[subitem]
-                  xml << "<#{subitem.question.odk_code}>"
+                  xml << "<#{Odk::DecoratorFactory.decorate(subitem).odk_code}>"
                   xml << instance[subitem]
-                  xml << "</#{subitem.question.odk_code}>"
+                  xml << "</#{Odk::DecoratorFactory.decorate(subitem).odk_code}>"
```

Comment:
Can you use a variable (e.g. `decorated_subitem`) to avoid doing this a bunch of times? Same below?

Reply:
I am pretty sure this is all going away with xml_submission